sdk/eidolon_ai_sdk/cpu/conversational_apu.py

```diff
@@ -247,12 +261,16 @@ def tool_event_stream():
                 else:
                     raise
 
-            message = ToolResponseMessage(
-                logic_unit_name=logic_unit_wrapper[0],
-                tool_call_id=tc.tool_call_id,
-                result=tool_stream.get_content() or "",
-                name=tc.name,
-            )
+            if isinstance(self.llm_unit, ToolCallLLMWrapper):
```

Comment:
another spot, possibly?

Reply:
But this one returns a different response type...

Reply:
yes same problem.

The new llm unit wrapper needs to transform tool calls to assistant messages for the llm (and the responses to user messages).

It should also translate it back on the way out, so the external message pattern looks the same. I actually didn't see that, but I imagine dave will take care of that when the logic gets pushed into the llm unit wrapper